my_game.py

```diff
@@ -450,12 +431,22 @@ def on_update(self, delta_time):
                 ufo_hit.destroy()
                 self.player_score += UFO_POINTS_REWARD
 
+        if any(self.player_sprite.collides_with_list(self.asteroid_list)):
+            if self.player_immortality_timer > 1:
```

Comment:
Put the timer in the _Player_ class. Add a method to the _Player_ called _is_imortal()_. That method should return a boolean which tells us if the player is immortal or not.

Reply:
Done and done! Changed it to Player.is_imortal